### PR TITLE
Optimize DrawTexturePro and DrawRectanglePro transformations

### PR DESCRIPTION
**Pull request by ChrisDill**
- Add check so rotation is only applied if rotation != 0.0f.
- Replace matrix usage by calculating the vertex data directly.

### CODE REVIEW

File: src/shapes.c [hunk 1]
```diff
@@ -619,31 +619,60 @@ void DrawRectanglePro(Rectangle rec, Vector2 origin, float rotation, Color color
...
+    Vector2 br = { 0 };
+    Vector2 tr = { 0 };
+    Vector2 tl = { 0 };
 
-    rlPushMatrix();
-        rlTranslatef(rec.x, rec.y, 0.0f);
-        rlRotatef(rotation, 0.0f, 0.0f, 1.0f);
-        rlTranslatef(-origin.x, -origin.y, 0.0f);
+    // Only calculate rotation if needed
+    if (rotation == 0.0f)
```
Comment (raysan5) [inline src/shapes.c:628]: Be careful, this check could result in false positives due to float resolution issues.

Comment (raysan5) [reply src/shapes.c:628]: Not sure how to solve it... I did a bit of research and it seems that the specific case for 0.0f should work ok... but no clear answer on this topic.

Comment (ChrisDill) [reply src/shapes.c:628]: It should work fine for functions that pass the rotation as 0.0f internally such as DrawTexture though I am not sure about situations where the user passes in rotation that is very close to 0.0f.

**Comment by raysan5** [comment]
@ChrisDill Excuse me for not merging yet, I'd like to do some tests first.

**Comment by raysan5** [comment]
@ChrisDill Thanks for the improvement! I did some tests and the gain is considerable! :)

**Pull request merged**